app/scripts/lib/experiments/grouping-rules/signup-password-confirm.js

```diff
@@ -0,0 +1,33 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define((require, exports, module) => {
+  'use strict';
+
+  const BaseGroupingRule = require('./base');
+
+  module.exports = class SignupPasswordConfirmGroupingRule extends BaseGroupingRule {
+    constructor () {
+      super();
+      this.name = 'signupPasswordConfirm';
+    }
+
+    choose (subject = {}) {
+      const sampleRate = SignupPasswordConfirmGroupingRule.sampleRate(subject.env);
+
+      return !! subject.uniqueUserId && this.bernoulliTrial(sampleRate, subject.uniqueUserId);
```

Comment:
> If we say 50% of people get the experiment, and of those people, half are put in treatment, and half in control, well, that's just 25% of users having the experiment enabled, right?

Not technically. In this example, 25% of people have the new feature enabled, but 50% of people are in the experiment. The control group is considered part of the experiment even though they don't see the new feature. 

Bucketing is done this way to ensure apples are compared to apples, all users in the control group must also be eligible for the treatment group. For this case, there are no restrictions for who can be in either group, but as a counter example, the SMS feature is only available to users in certain countries. Since we want to eliminate all possible causes of skew, users can only be in the SMS control group if they can also be in the SMS treatment group, meaning 1) they have to be in an eligible country, and 2) they have to have an active session. I got this wrong initially, I said the control group was taken from the general population, and this error skewed the metrics.

Explicitly labeling which bucket a user is a member of is necessary to track how an experiment such as SMS performs with a gradual release. We initially displayed the SMS screen to 2.5% of the eligible people people, saying the other 97.5% is automatically the "control" group is incorrect because not all of those 97.5% are eligible to be part of the control group. The control group has to come from the same eligible sample of users.

As far as bucketing users and the size of the experiment, we have been pretty conservative releasing features. A bug released to 2.5% of users is a lot easier to swallow than a bug released to 50%. We can rollback easily enough, but we can also point release to increase the % easily enough.